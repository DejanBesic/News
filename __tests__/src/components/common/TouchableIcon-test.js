import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import { TouchableIcon } from 'Components';
import { TouchableOpacity } from 'react-native-gesture-handler';

describe('TouchableIcon', () => {
  let props = {};
  const mock = jest.fn();
  beforeAll(() => {
    props = {
      name: 'menu',
      size: 20,
      onPress: mock,
      color: 'black',
    };
  });

  it('doesn\t crash without props', () => {
    const tree = renderer.create(<TouchableIcon onPress={mock} name="menu" />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    const tree = renderer.create(<TouchableIcon {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('triggers onPress correctly', () => {
    const tree = renderer.create(<TouchableIcon {...props} />);
    const container = tree.root.findByType(TouchableOpacity);
    renderer.act(() => {
      container.props.onPress();
    });
    expect(mock).toHaveBeenCalled();
  });
});
