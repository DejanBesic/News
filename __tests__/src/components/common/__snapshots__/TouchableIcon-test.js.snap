// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableIcon doesn	 crash without props 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
>
  <View
    accessible={true}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</RNGestureHandlerButton>
`;

exports[`TouchableIcon renders correctly with props 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
>
  <View
    accessible={true}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</RNGestureHandlerButton>
`;
